indexing-service/src/main/java/org/apache/druid/indexing/common/IngestionStatsAndErrorsTaskReportData.java

```diff
@@ -101,6 +106,13 @@ public long getSegmentAvailabilityWaitTimeMs()
     return segmentAvailabilityWaitTimeMs;
   }
 
+  @JsonProperty
+  @Nullable
+  public Map<String, Long> getRecordsProcessed()
```

Comment:
@adithyachakilam , just noticed this. This field is marked nullable for serialization. It should be marked nullable in the constructor too. Otherwise, we will not be able to deserialize older reports which do not have the field `recordsProcessed`. Please also add a test to verify this. You may include these changes in #16402.